lang/mcache/utils_test.go

```diff
@@ -26,3 +26,13 @@ func TestBsr(t *testing.T) {
 	assert.Equal(t, bsr((1<<10)-1), 9)
 	assert.Equal(t, bsr((1<<30)+(1<<19)+(1<<16)+(1<<1)), 30)
 }
+
+func BenchmarkBsr(b *testing.B) {
+	num := (1 << 30) + (1 << 19) + (1 << 16) + (1 << 1)
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		_ = bsr(num)
```

Comment:
Since num is a constant(and may be optimized by the compiler), looks like `bsr(num+i)` would be better?

Reply:
solved